samples/snippets/helpers.py

```diff
@@ -63,3 +65,21 @@ def wait_for_job_state(
         "\nTry increasing the timeout in sample test"
         f"\nLast recorded state: {response.state}"
     )
+
+def flaky_test_diagnostic(file_name, test_name, N=20):
+    
+    import pytest
```

Comment:
This is the only non-standard-library dependency for this helper function, so I am hiding the import in here so that pytest does not become a dependency for the whole module. I can also inject the module as an arg, or not worry about pytest as a dependency of the module at put it top-level.